### ISSUE TITLE
#2: implement common docker containers

### ISSUE THREAD

**Issue by tlamonthezie** [DARMA-tasking/workflows#2]
As in vt create a set of docker files for each common environment to test
Make it enough generic to be used in any project

**Comment by tlamonthezie** [comment]
List of base Docker files we want to analyze to extract some common dependencies and configurations :
-[] alpine-cpp.dockerfile
-[] develop.dockerfile
-[] ubuntu-20.04-gnu-openmpi-cpp.dockerfile
-[] ubuntu-clang-cpp.dockerfile
-[] ubuntu-gnu-cpp.dockerfile
-[] ubuntu-gnu-docs.dockerfile
-[] ubuntu-intel-oneapi-cpp.dockerfile
-[] ubuntu-nvidia-cpp.dockerfile

**Comment by cz4rs** [comment]
Where to push the images? Currently we are using `REPO: lifflander1/checkpoint` (at DockerHub). GitHub has it's `ghcr.io` that may be better integrated with GitHub infrastructure.

Example: https://github.com/orgs/Pressio/packages?repo_name=pressio.

@lifflander Do you have any preference in this regard?

**Comment by cz4rs** [comment]
TODO:
- drop the unused code and docker images
- clean up the PR so it can be reviewed
- the goal is to push new set of docker images (with a prefix to distinguish from "production" images)
- and start using them in some of the projects, `magistrate` seems like a good place to start
- reuse the vt testing from `build-and-test` to do the sanity testing before pushing

For magistrate integration:
- add `magistrate` testing before pushing new images (separate issue, do the general delivery of docker images first)

Long term:
- test the newly built images before pushing them to container repository (separate issue)

**Comment by cz4rs** [comment]
Things to consider:
- should we cancel all builds on single failure? for now no (they are run in parallel anyways), to be reconsidered in the future if needed

**Comment by cz4rs** [comment]
`ccache` - create a separate issue, might be unnecessary here (apart from setting a template for using it in the projects).
